### PR TITLE
adding ClassSimplexCriterion

### PR DESCRIPTION
**Pull request by soumith**
Reference: http://arxiv.org/abs/1506.08230

docs + unit test

Original Author: me
Reviewed By: @lvdmaaten

### CODE REVIEW

File: doc/criterion.md [hunk 1]
```diff
@@ -162,6 +163,52 @@ loss(x, class) = weights[class] * (-x[class] + log(\sum_j exp(x[j])))
...
+The following is a code fragment showing how to make a gradient step given an input `x`, a desired output `y` (an integer `1` to `n`, in this case `n = 30` classes), a network `mlp` and a learning rate `learningRate`:
+
+```lua
+nInput = 10
+nClasses = 30
+nHidden = 100
+mlp = nn.Sequential()
+mlp:add(nn.Linear(nInput, nHidden)):add(nn.ReLU())
+mlp:add(nn.NormalizedLinearNoBias(nHidden, nClasses))
+mlp:add(nn.L2Normalize())
```
Comment (fmassa) [inline doc/criterion.md]: Do you need this `L2Normalize` in here ? Can't it be replaced by `nn.Normalize(2)` ?

Comment (soumith) [inline doc/criterion.md]: yea now i can replace it with nn.Normalize(2). Back when I wrote it, there was no nn.Normalize

Comment (soumith) [inline doc/criterion.md]: fixed via https://github.com/torch/nn/commit/0d4ae479755f0f1547c1e4bcddf8ea198bf50063
